http-server/src/main/java/io/airlift/http/server/HttpServer.java

```diff
@@ -139,7 +144,19 @@ public HttpServer(
             log.info("Virtual threads support is enabled");
             threadPool.setVirtualThreadsExecutor(executor);
         }
-        server = new Server(threadPool);
+
+        int maxBufferSize = toIntExact(max(
+                toSafeBytes(config.getMaxRequestHeaderSize()),
+                toSafeBytes(config.getMaxResponseHeaderSize())));
+
+        ByteBufferPool byteBufferPool = new ArrayByteBufferPool.Quadratic(
+                0, // min capacity
+                maxBufferSize == -1 ? DEFAULT_MAX_CAPACITY : maxBufferSize, // max capacity
+                Integer.MAX_VALUE, // maximum number of buffers
```

Comment:
drop the  `// maximum number of buffers` comment. It is actually max bucket size - which counts bytes - not number of objects.